### PR TITLE
Better command line tool

### LINKED ISSUE #585: Artifact as Subject does not get picked up during rating calculation
Below error

```
java -jar target\fosstars-github-rating-calc.jar -g com.fasterxml.jackson.core:jackson-databind:2.9.1 -t {github_token} --rating oss-artifact-security
[+] Okay, we have a GitHub token, let's try to use it
[+] Start with com.fasterxml.jackson.core:jackson-databind:2.9.1
[+] SCM is http://github.com/FasterXML/jackson-databind
[+] Let's gather info and calculate a rating for:
[+]   https://github.com/FasterXML/jackson-databind
[+] Figuring out if the project belongs to the Eclipse Software Foundation ...
[+] Figuring out how the project uses CodeQL ...
[+] Cloning https://github.com/FasterXML/jackson-databind ...
[!] Holy Moly, ReleaseInfoFromMaven data provider failed!
[!] The last thing that it said was
java.lang.IllegalArgumentException: Oh no! Expected com.sap.oss.phosphor.fosstars.model.subject.oss.MavenArtifact
        at com.sap.oss.phosphor.fosstars.model.Subject.cast(Subject.java:69)
        at com.sap.oss.phosphor.fosstars.data.artifact.ReleaseInfoFromMaven.update(ReleaseInfoFromMaven.java:57)
        at com.sap.oss.phosphor.fosstars.data.artifact.ReleaseInfoFromMaven.update(ReleaseInfoFromMaven.java:27)
        at com.sap.oss.phosphor.fosstars.tool.github.SingleRatingCalculator.calculateFor(SingleRatingCalculator.java:92)
        at com.sap.oss.phosphor.fosstars.tool.github.Application.processUrl(Application.java:411)
        at com.sap.oss.phosphor.fosstars.tool.github.Application.process(Application.java:502)
        at com.sap.oss.phosphor.fosstars.tool.github.Application.process(Application.java:462)
        at com.sap.oss.phosphor.fosstars.tool.github.Application.processGav(Application.java:441)
        at com.sap.oss.phosphor.fosstars.tool.github.Application.run(Application.java:357)
        at com.sap.oss.phosphor.fosstars.tool.github.Application.main(Application.java:165)
[!] But we don't give up!
[!] Holy Moly, ReleaseInfoFromNpm data provider failed!
[!] The last thing that it said was
java.lang.IllegalArgumentException: Oh no! Expected com.sap.oss.phosphor.fosstars.model.subject.oss.NpmArtifact
        at com.sap.oss.phosphor.fosstars.model.Subject.cast(Subject.java:69)
        at com.sap.oss.phosphor.fosstars.data.artifact.ReleaseInfoFromNpm.update(ReleaseInfoFromNpm.java:51)
        at com.sap.oss.phosphor.fosstars.data.artifact.ReleaseInfoFromNpm.update(ReleaseInfoFromNpm.java:28)
        at com.sap.oss.phosphor.fosstars.tool.github.SingleRatingCalculator.calculateFor(SingleRatingCalculator.java:92)
        at com.sap.oss.phosphor.fosstars.tool.github.Application.processUrl(Application.java:411)
        at com.sap.oss.phosphor.fosstars.tool.github.Application.process(Application.java:502)
        at com.sap.oss.phosphor.fosstars.tool.github.Application.process(Application.java:462)
        at com.sap.oss.phosphor.fosstars.tool.github.Application.processGav(Application.java:441)
        at com.sap.oss.phosphor.fosstars.tool.github.Application.run(Application.java:357)
        at com.sap.oss.phosphor.fosstars.tool.github.Application.main(Application.java:165)
[!] But we don't give up!
[+] Figuring out how the project uses LGTM ...
```

Things to do:
- Make sure that Artifact as a Subject gets picked up by data provider.

### LINKED ISSUE #588: Refactor the command-line tool
The `Application` class is too big. It becomes difficult to maintain it. It causes issues like #585. In addition, the `tool.github` package contains classes that are not really about GitHub. Let's refactor it.

### PR DESCRIPTION
**Pull request by artem-smotrakov**
Here is a list of main updates:

- Spring the `Application` class to several handlers that run calculation of specific ratings.
- Updated the CLI to support calculating security ratings for artifacts (Maven and NPM).

Closes #585
Closes #588

### CODE REVIEW

File: src/main/java/com/sap/oss/phosphor/fosstars/tool/OssArtifactSecurityRatingHandler.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        project.orElse(NO_PROJECT));
+
+    calculator().calculateFor(artifact);
+
+    if (!artifact.ratingValue().isPresent()) {
+      throw new IOException("Could not calculate a rating!");
+    }
+
+    Arrays.stream(createFormatter("text").print(artifact).split("\n")).forEach(logger::info);
+    logger.info("");
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/oss/phosphor/fosstars/tool/OssArtifactSecurityRatingHandler.java:76]: This is not required.

Comment (artem-smotrakov) [reply src/main/java/com/sap/oss/phosphor/fosstars/tool/OssArtifactSecurityRatingHandler.java:76]: This is a line separator to make the CLI's output nicer. So, let's please keep it.

File: src/main/java/com/sap/oss/phosphor/fosstars/tool/AbstractHandler.java [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+  @Override
+  public Handler set(SubjectValueCache cache) {
+    requireNonNull(cache, "Oops! Cache is null!");
+    this.cache = cache;
+    return this;
+  }
+
+  @Override
+  public void close() {
+
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/oss/phosphor/fosstars/tool/AbstractHandler.java:194]: Could we add some exception here?

**Review by sourabhsparkala — APPROVED** [review]
👍 
I tested the CLI as well. Looks good.

File: src/main/java/com/sap/oss/phosphor/fosstars/tool/AbstractHandler.java (hunk 2 above)
Comment (artem-smotrakov) [reply src/main/java/com/sap/oss/phosphor/fosstars/tool/AbstractHandler.java:194]: I was going to use `AutoCloseable` to do a little cleanup in the end, but it is not actually used. I'll just drop it and remove this method.

**Comment by artem-smotrakov** [comment]
> I tested the CLI as well. Looks good.

Thanks for the testing and review @sourabhsparkala ! Please have a look again -- I've merged the updates from #595.

**Review by sourabhsparkala — APPROVED** [review]
👍

**Pull request merged**